osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -136,39 +136,42 @@ private void createSsldPendingFirmwareUpdateRecord(final CorrelationIds ids, fin
+            final FirmwareModuleType firmwareModuleType = FirmwareModuleType.valueOf(firmwareModule.get());
 
-            SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = new SsldPendingFirmwareUpdate(true,
-                    firmwareModuleType, firmwareVersion, "CORE", "1.0", ids.getOrganisationIdentification());
+            SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = new SsldPendingFirmwareUpdate(
+                    ids.getDeviceIdentification(), firmwareModuleType, firmwareVersion,
+                    ids.getOrganisationIdentification(), ids.getCorrelationUid());
             ssldPendingFirmwareUpdate = this.ssldPendingFirmwareUpdateRepository.save(ssldPendingFirmwareUpdate);
 
-            ssld.setSsldPendingFirmwareUpdate(ssldPendingFirmwareUpdate);
-            this.ssldRepository.save(ssld);
-
-            LOGGER.info("Saved pending fimware update record for SSLD: {}, {}", ssld.getDeviceIdentification(),
+            LOGGER.info("Saved pending fimware update record for SSLD: {}, {}", ids.getDeviceIdentification(),
                     ssldPendingFirmwareUpdate);
         } catch (final Exception e) {
             LOGGER.error("Caugth exception when creating pending firmware update record for SSLD: {}",
-                    ssld.getDeviceIdentification(), e);
+                    ids.getDeviceIdentification(), e);
         }
     }
 
-    public void handleSsldPendingFirmwareUpdate(final String organisationIdentification,
-            final String deviceIdentification, final String correlationUid) {
+    public void handleSsldPendingFirmwareUpdate(final String deviceIdentification) {
+
+        final SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = this.ssldPendingFirmwareUpdateRepository
+                .findByDeviceIdentification(deviceIdentification);
+
+        if (ssldPendingFirmwareUpdate != null) {
+            final String organisationIdentification = ssldPendingFirmwareUpdate.getOrganisationIdentification();
+            final String correlationUid = ssldPendingFirmwareUpdate.getCorrelationUid();
 
-        final Ssld ssld = this.ssldRepository.findByDeviceIdentification(deviceIdentification);
-        final SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = ssld.getSsldPendingFirmwareUpdate();
-        if (ssldPendingFirmwareUpdate.hasPendingFirmwareUpdate()) {
             LOGGER.info(
                     "Handling SSLD pending firmware update for device identification: {}, organisation identification: {} and correlation UID: {}.",
                     deviceIdentification, organisationIdentification, correlationUid);
 
-            ssldPendingFirmwareUpdate.setCorrelationUid(correlationUid);
-            this.ssldPendingFirmwareUpdateRepository.save(ssldPendingFirmwareUpdate);
-
             try {
+                final int messagePriority = MessagePriorityEnum.DEFAULT.getPriority();
                 this.getFirmwareVersion(organisationIdentification, deviceIdentification, correlationUid,
```

Comment:
As discussed, consider adding a (configurable) delay to make sure the device registration process is finished before the get firmware request is sent to the device.